pkg/common/utils/utils.go

```diff
@@ -154,15 +163,26 @@ func GetApplicationIDFromPod(pod *v1.Pod) string {
 		}
 	}
 
-	// Application ID can be defined in annotation
```

Comment:
The order to get 'app-id' from pod:

Before this PR:
1. Annotation: constants.AnnotationApplicationID
2. Label: constants.LabelApplicationID
3. Label: constants.SparkLabelAppID

After this PR
1. **Label: constants.CanonicalLabelApplicationID (New)**
2. Label: constants.LabelApplicationID
3. Label: constants.SparkLabelAppID
4. **Annotation: constants.AnnotationApplicationID(Move to the last, label first)**